### DISCUSSION TITLE
[Rails] Slim or ERB?

### DISCUSSION THREAD

**Discussion by malparty** [nimblehq/rails-templates#484] [Polls]
The CRUD addon will include many views. In order to keep the maintenance efforts low, we must choose only 1 of these 2 templating engines.

From my experience and what I've seen, both work well. So this is an `opinionated` area, but we must make a decision 🤗

> [!NOTE] 
> The discussion will be closed on Friday 24th evening. 🌔

**Comment by malparty** [comment]
My personal experience with ERB in a Team project:
- There is NO auto-format extension for VS Code (this leads to some indentation issues but also some HTML closing tags missing :o )
- The usage of `"` double quotes for HTML code and `'` single quotes for Ruby code in the same file was very confusing for some team members

I do not have experience with Slim though, thus the pool to ask for more insights 🙏

**Comment by longnd** [comment]
I prefer slim, it looks cleaner & less verbal compared to ERB.

**Comment by andyduong1920** [comment]
I prefer Slim too, less typing, less typo error 😆 - no need to think about the closing HTML tag

**Comment by byhbt** [comment]
I chose ERB because I prefer the native approach of the Rails framework. Since this scaffold is built upon the Rails framework. We can switch to Slim depending on the project's requirements.

**Reply by malparty** [reply]
Thanks for your sharing Thanh 🙏 

> We can switch to Slim depending on the project's requirements.

At the moment this decision will only impact the CRUD-addon (which will ship with custom layouts and view templates). This means:

- you can still select `ERB` when you do not select the `CRUD-addon`
- if you select the crud-addon (and if we go for Slim), it will be tedious to translate all the layouts and view templates back to ERB

That said, it is very nice to hear reasons in **favor** of ERB (Rails default, most plugins ship with ERB generators, most tutorials are using ERB, the Rails guides are all in ERB, etc...).

Another reason mentioned by Hoang (during the coffee break ☕ ) is that Slim makes it harder to copy/paste HTML snippets from the internet, as you need to translate them (though online converters can do the job now, e.g. https://tools.kaklabs.com/slim-converter.html ).

**Reply by byhbt** [reply]
Yup, so if we implement the CRUD-addon using `ERB`, when we generate the `views` for a business object (e.g: Team, Branch, Company...) 

Given the existing project is using :
- ERB -> That's fine
- Slim -> We can use the tool you mentioned for converting it to `Slim` syntax.

**Reply by malparty** [reply]
My idea for the CRUD addon is to make the arbitration, not to find some workarounds to keep it generic. If it's too generic, it will be too complex to use and it won't benefit the company.
That's why I'm not looking for a solution to make it compatible for both template engines, I'm looking for what template engie to use 😇

**Comment by malparty** [comment]
- 8 votes in favor of Slime
- 1 vote in favor of ERB

The CRUD add-on will be implemented with the Slim template engine 🚀 
Thanks for your participation :)

![image](https://github.com/nimblehq/rails-templates/assets/77609814/472de68e-3b9c-4b0c-9510-bba516e7f345)
